docker-compose.yml

```diff
@@ -1,11 +1,37 @@
-app:
-  mem_limit: 300m
-  memswap_limit: 1g
-  env_file:
-    .env
-  build: .
-  ports:
-    - '3000:8000'
-  volumes:
-    - .:/home/nodejs/app
-    - /home/nodejs/app/node_modules
+version: '2'
+
+services:
+  mongodb:
+    container_name: ya-mongo-db
+    image: alexpunct/mongo:3.4
+    volumes:
+      - mongostorage:/data/db
+    ports:
+      - "${MONGODB_EXPOSED_PORT}:27017"
+    environment:
+      - MONGO_DATA_DIR=/data/db
+      - MONGO_LOG_DIR=/dev/null
+      - MONGODB_ADMIN_USER=${MONGODB_ADMIN_USER}
+      - MONGODB_ADMIN_PASS=${MONGODB_ADMIN_PASS}
+      - MONGODB_APPLICATION_DATABASE=${MONGODB_APPLICATION_DATABASE}
+      - MONGODB_APPLICATION_USER=${MONGODB_APPLICATION_USER}
+      - MONGODB_APPLICATION_PASS=${MONGODB_APPLICATION_PASS}
+
+  api:
+    container_name: ya-server
+    build: .
+    restart: always
+    mem_limit: 300m
+    ports:
+      - "${REST_API_EXPOSED_PORT}:3000"
+    links:
+      - mongodb
+    volumes:
+      - ./:/home/nodejs/app
+      - /home/nodejs/app/node_modules
```

Comment:
Not sure about this line

Reply:
why?

Reply:
The thing is, node_modules would be installed specific to the docker os environment, whereas you might use them (?) in your host environment, which might cause random errors